### PR TITLE
Multi module + Coroutines

### PR DESCRIPTION
**Pull request by BavovnaIsTaken**
Moved project to multi module structure, added ethree-kotlin-coroutines wrapper to easily work with Kotlin's coroutines.

### CODE REVIEW

File: common/src/main/java/com/android/virgilsecurity/common/Exceptions.kt [hunk 1]
```diff
@@ -31,7 +31,7 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package com.virgilsecurity.ethree.data.exception
+package com.android.virgilsecurity.common
```
Comment (andrii-iakovenko) [inline common/src/main/java/com/android/virgilsecurity/common/Exceptions.kt]: Maybe `com.android.virgilsecurity.common.exceptions` would be better

File: ethree/build.gradle [hunk 2]
```diff
@@ -9,22 +9,27 @@ apply plugin: 'maven'
 apply plugin: 'org.jetbrains.dokka'
 
 group 'com.virgilsecurity'
-version '0.1.0-aplha'
+version '0.1.1-alpha'
```
Comment (andrii-iakovenko) [inline ethree/build.gradle]: Common has `0.1.1-rc` version, but ethree is `0.1.1alpha`

File: ethreeCoroutines/build.gradle [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    id 'maven-publish'
+    id 'signing'
+}
+
+apply plugin: 'kotlin'
+apply plugin: 'maven'
+apply plugin: 'org.jetbrains.dokka'
+
+group 'com.virgilsecurity'
+version '0.1.0-alpha'
```
Comment (andrii-iakovenko) [inline ethreeCoroutines/build.gradle]: Version is differ from `common`

File: ethreeCoroutines/src/main/java/com/android/virgilsecurity/ethreecoroutines/extensions/CoroutineScopeExt.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.android.virgilsecurity.ethreecoroutines.extensions
```
Comment (andrii-iakovenko) [inline ethreeCoroutines/src/main/java/com/android/virgilsecurity/ethreecoroutines/extensions/CoroutineScopeExt.kt]: Why not `com.virgilsecurity.ethreecoroutines.android.extensions` ?

File: testscoroutines/src/test/java/com/android/virgilsecurity/testscoroutines/ExampleUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (andrii-iakovenko) [inline testscoroutines/src/test/java/com/android/virgilsecurity/testscoroutines/ExampleUnitTest.java]: We don't need this test

File: ethreeCoroutines/src/main/java/com/android/virgilsecurity/ethreecoroutines/extensions/CoroutineScopeExt.kt (hunk 4 above)
Comment (BavovnaIsTaken) [reply ethreeCoroutines/src/main/java/com/android/virgilsecurity/ethreecoroutines/extensions/CoroutineScopeExt.kt]: but we have 'com.android.virgilsecurity.ethreecoroutines.extensions' package, so it's right, isn't it?

**Review by andrii-iakovenko — DISMISSED** [review]

File: ethree/build.gradle [hunk 6]
```diff
@@ -9,7 +9,7 @@ apply plugin: 'maven'
 apply plugin: 'org.jetbrains.dokka'
 
 group 'com.virgilsecurity'
-version '0.1.1-alpha'
+version '0.2.0'
```
Comment (andrii-iakovenko) [inline ethree/build.gradle:12]: Why not 0.1.1?

**Review by andrii-iakovenko — APPROVED** [review]

**Pull request merged**